sound/soc/sof/intel/hda-dsp.c

```diff
@@ -367,7 +367,7 @@ int hda_dsp_set_power_state(struct snd_sof_dev *sdev,
 	dev_vdbg(bus->dev, "D0I3C updated, register = 0x%x\n",
 		 snd_hdac_chip_readb(bus, VS_D0I3C));
 
-	if (d0_substate == SOF_DSP_D0I0)
+	if (state == SOF_DSP_D0I0)
 		flags = HDA_PM_PPG;/* prevent power gating in D0 */
 	else
 		flags = HDA_PM_NO_DMA_TRACE;/* disable DMA trace in D0I3*/
```

Comment:
For D3, we should not send PM_GATE to FW.

Reply:
the set_power_state() function would really ony be used for D0i0 or D0i3. For D3, we'd never call this. We just pull the plug isnt it?